realm/src/androidTest/java/io/realm/RealmJsonTest.java

```diff
@@ -121,6 +123,23 @@ public void testCreateObjectFromJson_dateAsString() throws JSONException {
         assertEquals(new Date(1000), obj.getColumnDate());
     }
 
+    public void testCreateObjectFromJson_dateAsStringTimeZone() throws JSONException {
+        JSONObject json = new JSONObject();
+
+        // Oct 03 2015 14:15
+        json.put("columnDate", "/Date(1443854733376+0800)/");
+
+        testRealm.beginTransaction();
+        testRealm.createObjectFromJson(AllTypes.class, json);
+        testRealm.commitTransaction();
+
+        AllTypes obj = testRealm.allObjects(AllTypes.class).first();
+        Calendar cal = GregorianCalendar.getInstance();
+        cal.set(2015,Calendar.OCTOBER, 03, 14, 45, 33);
+        Date convDate = obj.getColumnDate();
+        assertTrue(convDate.getTime()-cal.getTime().getTime()<1000);
```

Comment:
This assert could is not very readable, I'm guessing you do it because we truncate dates to seconds, but if you do

````
cal.set(Calendar.MILLISECOND, 0)
//...
assertEquals(cal.getTime(), convDate)
```

It will become much more readable.